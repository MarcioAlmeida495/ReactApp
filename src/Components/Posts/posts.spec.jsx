import { render, screen } from "@testing-library/react"
import { Posts } from ".";

const props = {
    posts:[{
        id: 1,
        title: 'title 1',
        body: 'body1',
        cover: 'img/img1.png'
    },{
        id: 2,
        title: 'title 3',
        body: 'body2',
        cover: 'img/img2.png'
    },{
        id: 3,
        title: 'title 3',
        body: 'body3',
        cover: 'img/img3.png'
    }]
}

describe('<Posts />', () => {
    it('should render posts', ()=> {
        render(<Posts {...props} />);
        expect(screen.getAllByRole('heading', {name: /title/i})).toHaveLength(3);
        //debug();
    })
    it('should match a snapshot', ()=> {
        const {container} = render(<Posts {...props} />);
        expect(container).toMatchSnapshot();
    })
})
