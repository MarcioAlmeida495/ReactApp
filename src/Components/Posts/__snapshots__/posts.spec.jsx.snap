// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match a snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="title 1"
        src="img/img1.png"
      />
      <div
        class="post-content"
      >
        <h3>
          title 1
        </h3>
        <p>
          body1
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title 3"
        src="img/img2.png"
      />
      <div
        class="post-content"
      >
        <h3>
          title 3
        </h3>
        <p>
          body2
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title 3"
        src="img/img3.png"
      />
      <div
        class="post-content"
      >
        <h3>
          title 3
        </h3>
        <p>
          body3
        </p>
      </div>
    </div>
  </div>
</div>
`;
